### PR TITLE
[SPARK-31476][SQL] Add an ExpressionInfo entry for EXTRACT

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR intends to add an ExpressionInfo entry for EXTRACT for better documentations.
This PR comes from the comment in https://github.com/apache/spark/pull/21479#discussion_r409900080

### Why are the changes needed?

To make SQL documentations complete.

### Does this PR introduce any user-facing change?

Yes, this PR updates the `Spark SQL, Built-in Functions` page.

### How was this patch tested?

Run the example tests.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Could you regenerate `date.sql`, too?
```
sbt.ForkMain$ForkError: org.scalatest.exceptions.TestFailedException: postgreSQL/date.sql
Expected "struct<[date_part('EPOCH',] DATE '1970-01-01'):...", but got "struct<[extract('EPOCH' ```

**Comment by maropu** [comment]
Ah, yes. I'm updating now. Thanks for the check, anyway.

**Comment by maropu** [comment]
retest this please

**Comment by dongjoon-hyun** [comment]
cc @gatorsmile

**Comment by dongjoon-hyun** [comment]
Merged to master/3.0. Thank you, @maropu .

**Comment by maropu** [comment]
Thanks for the quick response, @dongjoon-hyun ! cc: @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala [hunk 1]
```diff
@@ -423,6 +423,7 @@ object FunctionRegistry {
     expression[MakeTimestamp]("make_timestamp"),
     expression[MakeInterval]("make_interval"),
     expression[DatePart]("date_part"),
+    expression[Extract]("extract"),
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:426]: one side effect is now we support `extract(field, source)` other than `extract(field from source)`. Not a big deal but better if we can avoid exposing more APIs,

Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:426]: > Not a big deal but better if we can avoid exposing more APIs

Yea, +1 .

btw, its better to add tests for the case `extract(field, source)`?

Comment (cloud-fan) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:426]: Yea if we decide to support it.

Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:426]: ok, I'll do follow-up later.

Comment (maropu) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:426]: I opened https://github.com/apache/spark/pull/28276

**Pull request closed without merging**